mbtools/models.py

```diff
@@ -87,16 +86,21 @@ def html_elements(self):
             contents = page.xpath("contents")
 
             location = \
-                f'Lesson: {lesson_name} Page_id={page_id}'\
+                f'Lesson: {lesson_name} (page id={page_id}) ' \
                 f'Page Title: {page_title}'
             elements.append(MoodleHtmlElement(contents[0], location))
 
-            if len(page.xpath("answers")) > 0:
-                answer_texts = page.xpath("answers/answer_text")
-                for answer in answer_texts:
-                    answer_location = f'{location} Answer: {answer.text}'
+            for answer in page.xpath("answers/answer"):
+                answer_format = answer.xpath("answerformat")[0]
+                # We need to check the answer format to filter out things
+                # like buttons which are also serialized as answers but are
+                # not HTML
+                if answer_format.text == "1":
+                    answer_text = answer.xpath("answer_text")[0]
+                    answer_location = \
+                        f'{location} (answer id: {answer.attrib["id"]})'
```

Comment:
So what this is trying to do is separate actual lesson page problems where there is HTML content such as on [this page](https://raiselearning.org/mod/lesson/view.php?id=418&pageid=441) from cases like [this page](https://raiselearning.org/mod/lesson/view.php?id=418&pageid=387) where the navigation buttons as still serialized as answers in the underlying XML (I have not dug into the actual Moodle code to understand where / why -- this is purely based on inspection of the XML it generatess). 

The schema for a question answer on the former:

```xml
<answer id="849">
            <jumpto>389</jumpto>
            <grade>0</grade>
            <score>0</score>
            <flags>0</flags>
            <timecreated>1646156371</timecreated>
            <timemodified>0</timemodified>
            <answer_text>&lt;p dir="ltr" style="text-align: left;"&gt;\( 3n \)&lt;br&gt;&lt;/p&gt;</answer_text>
            <response>&lt;p dir="ltr" style="text-align: left;"&gt;Incorrect. Let’s try again a different way: Each student gets 2 slices of pizza. Multiply the cost of one slice&amp;nbsp;\( (3n) \) by 2. Also, add the delivery fee of $8. The answer is \( (2)3n  + 8 \).&lt;br&gt;&lt;/p&gt;</response>
            <answerformat>1</answerformat>
            <responseformat>1</responseformat>
            <attempts>
            </attempts>
          </answer>
```

The schema for one of the buttons on the latter:

```xml
<answer id="738">
            <jumpto>-40</jumpto>
            <grade>0</grade>
            <score>0</score>
            <flags>0</flags>
            <timecreated>1636583392</timecreated>
            <timemodified>0</timemodified>
            <answer_text>Previous: 1.1.1: Activity</answer_text>
            <response>$@NULL@$</response>
            <answerformat>0</answerformat>
            <responseformat>0</responseformat>
            <attempts>
            </attempts>
          </answer>
```

You'll notice that the only way to tell these apart is based upon the value of `<answerformat>`, and we need to tell them apart because otherwise we'll get a **TON** of false positive noise from buttons being unnested HTML content.